### ISSUE TITLE
Roslyn incorrectly report some API as `Not Available ⚠` in IntelliSense list for mulit targetframeworks projects

### ISSUE THREAD

**Issue by yyjdelete** [dotnet/roslyn#58478]
**Version Used**: 
VS2022 17.1.0 preview1.1

**Steps to Reproduce**:

1. Create an new console app, and change it to multi target frameworks with `net6.0;net472`
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>net6.0;net472</TargetFrameworks>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <LangVersion>latest</LangVersion>
  </PropertyGroup>

</Project>
```
2. Try to autocomplete `ProcessStartInfo` with the below code
```cs
using System.Diagnostics;

Process.Start(new Pro { });//<--Try to autocomplete `ProcessStartInfo` here and see it in IntelliSense list
```
3. 

**Expected Behavior**:
The API should be available for both target framework.

**Actual Behavior**:
Roslyn report `ProcessStartInfo` is not avilable in net472
![image](https://user-images.githubusercontent.com/1819074/147312290-c2b2a363-dffc-4df0-9b9f-4868952f1c7c.png)

**Comment by genlu** [comment]
Looks like this was fixed

![image](https://user-images.githubusercontent.com/788783/192360474-2cd89ba9-4220-49dd-a62d-42fbf42ba2cd.png)
